gc/base/SweepHeapSectioning.cpp

```diff
@@ -309,16 +309,14 @@ MM_SweepHeapSectioning::update(MM_EnvironmentBase* env)
 		}
 
 		/* link the new array into the list of arrays */
-		newArray->_next = _head;
-		_head = newArray;
+		Assert_MM_true(NULL != _tail);
+		Assert_MM_true(NULL == _tail->_next);
+
+		_tail->_next = newArray;
+		_tail = newArray;
```

Comment:
while at this method/file, fix some coding standard issues:
- put declaration and initialization at the same line for totalChunkCount, newArray, chunk or anything else that you notice
- fix (add) spacing between for and '(' , and around some binary operators like '<', '=', ....